c2rust-transpile/src/translator/mod.rs

```diff
@@ -3411,7 +3394,35 @@ impl<'c> Translation<'c> {
                     // We want to decay refs only when function is variadic
                     ctx.decay_ref = DecayRef::from(is_variadic);
 
-                    let args = self.convert_exprs(ctx.used(), args)?;
+                    let mut args = self.convert_exprs(ctx.used(), args)?;
+
+                    // the C variadics feature requires us to call `as_va_list` on a `VaListImpl`
+                    // when calling a function expecting to receive an object that is binary
+                    // compatible with C's `va_list`.
+                    if let Some(CTypeKind::Function(_, params, _, _, _)) = fn_ty {
+                        // look for parameters of type `va_list`
+                        let positions: Vec<usize> = params
+                            .iter()
+                            .enumerate()
+                            .filter_map(|(pos, param)| {
+                                if self.ast_context.is_pointer_to_va_list(param.ctype)
+                                { Some(pos) } else { None }
```

Comment:
I'm not sure this is how `rust-fmt` would indent it. You could try running it on this file, but that's probably going to produce a ton of other changes.